### DISCUSSION TITLE
how does the business system receive or send Signal communication?

### DISCUSSION THREAD

**Discussion by landonzeng** [elsa-workflows/elsa-core#1465] [Q&A] [answered]
I want to deploy it separately as Elsa Dashboard + Server as a workflow service, and then other business systems call the workflow designed by Elsa dashboard via http. When using SendSignal and Signal Received, how does the business system receive or send Signal communication?

**Comment by sfmskywalker** [comment, marked as answer]
The Elsa signaling mechanism was designed only to send signals to workflows, not to business systems.

For that, a better fit would be the *Event Received*, *RunTask* and *Send Command* activities found in `Elsa.Activities.Conductor`.

I haven't documented these yet, but they basically allow you to implement a conductor-style workflow where your workflow orchestrates tasks and events.

**RunTask** activities allow you to run a task in your business system, and your business system can send events to the workflow to start or resume execution of the workflow by triggering **EventReceived** activities.

As of this writing, the way for the workflow to notify your business system to execute a given task is by letting your business system expose a webhook endpoint that the workflow server will invoke. But the goal is to make this pluggable, meaning that there will be other transports available in the future such as AMQP, SignalR and any custom implementation you might want to use.

**Reply by landonzeng** [reply]
Thank you for your answer
